src/ui/public/agg_types/metrics/top_hit.js

```diff
@@ -0,0 +1,118 @@
+    name: 'top_hits',
+    title: 'Top Hit',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Last' : 'First';
+      return `${prefix} ${aggConfig.params.field.displayName}`;
+    },
+    params: [
+      {
+        name: 'field',
+        onlyAggregatable: false,
+        showAnalyzedWarning: false,
+        filterFieldTypes: function (vis, value) {
+          if (vis.type.name === 'table' || vis.type.name === 'metric') {
+            return true;
+          }
+          return value === 'number';
+        },
+        write(agg, output) {
+          const field = agg.params.field;
+          output.params = { size: 1 };
+
+          if (field.scripted) {
+            output.params.script_fields = {
+              [ field.name ]: {
+                script: {
+                  inline: field.script,
+                  lang: field.lang
+                }
+              }
+            };
+          } else {
+            // do not try to get the doc_values for IP fields, since it is
+            // an internal representataion of the IP and so of no use for display.
+            if (field.type !== 'ip' && field.doc_values) {
+              output.params.docvalue_fields = [ field.name ];
+            }
+            output.params._source = field.name;
+          }
+        }
+      },
+      {
+        name: 'sortField',
+        type: 'field',
+        editor: null,
+        filterFieldTypes: [ 'number', 'date', 'ip',  'string' ],
+        default: function (agg) {
+          return agg.vis.indexPattern.timeFieldName;
+        },
+        write: _.noop // prevent default write, it is handled below
+      },
+      {
+        name: 'sortOrder',
+        type: 'optioned',
+        default: 'desc',
+        editor: topSortEditor,
+        options: [
+          { display: 'Descending', val: 'desc' },
+          { display: 'Ascending', val: 'asc' }
+        ],
+        write(agg, output) {
+          const sortField = agg.params.sortField;
+          const sortOrder = agg.params.sortOrder;
+
+          if (sortField.scripted) {
+            output.params.sort = [
+              {
+                _script: {
+                  script: {
+                    inline: sortField.script,
+                    lang: sortField.lang
+                  },
+                  type: sortField.type,
+                  order: sortOrder.val
+                }
+              }
+            ];
+          } else {
+            output.params.sort = [
+              {
+                [ sortField.name ]: {
+                  order: sortOrder.val
+                }
+              }
+            ];
+          }
+        }
+      }
+    ],
+    getValue(agg, bucket) {
+      const hits = _.get(bucket, `${agg.id}.hits.hits`);
+      if (!hits || !hits.length) {
+        return null;
+      }
+      const path = agg.params.field.name;
+      let values = agg.vis.indexPattern.flattenHit(hits[0])[path];
```

Comment:
A couple notes in regards to flattening/formatting after I ran some tests:

- [ ] I think you'll want to use [formatHit or formatField](https://github.com/elastic/kibana/blob/0b877f63135955944677231c6661653246f0bf0e/src/ui/public/index_patterns/_index_pattern.js#L168-L168) instead of just flattenHit. For array fields this will give us a nice comma delimited list like on the Discover page instead of a stringified array. You might need to test and make sure this doesn't apply formatting twice when a custom formatter is specified though.

- [ ] Objects in arrays aren't handled properly in _source, even though ES is returning all the info we need. We specifically mention that we don't handle this scenario well on other pages, like Discover. What happens is for a field like `foo.bar`, where `foo` is an array of objects and `bar` is a property of those objects, ES will return something like:

```
"_source": {
  "foo": [
    {
      "bar": "1"
    },
    {
      "bar": "2"
    }
  ]
}
```

I think the best solution would be to enhance flattenHit/formatHit/formatField to handle this better everywhere, but that feels like major scope creep for this PR. For now I think it would be sufficient to do what discover does, and displayed the stringified array currently returned from `formatHit`. So in the example above, formatHit would return an object with a `foo` field, and we would just display the contents of that `foo` field.